basic_algs/src/main.rs

```diff
@@ -0,0 +1,66 @@
+fn min(array: &[f64; 10]) -> f64 {
+    let mut cur_min = array[0];
+    for element in array {
+        if cur_min > *element {
+            cur_min = *element
+        }
+    }
+    return cur_min
+}
+
+fn is_prime(n: &i32) -> bool {
```

Comment:
Предлагаю для примитивных числовых типов здесь и далее использовать передачу по значению:
```suggestion
fn is_prime(n: i32) -> bool {
```
`i32` реализует трейт Copy, а значит тривиально копируется. Передача по значению решает следующие проблемы:
1. Обилие разыменований ссылок в коде. Лишние `*` создают неудобство для читателя.
2. Необходимость вызывать функцию с `&` перед аргументом. Также дополнительная мелкая деталь, делающая код чуть более сложным для чтения.
3. Передача по ссылке может быть менее эффективна для типов, размер которых меньше или равен размеру ссылки (т.е. 8 байт на 64-битных системах). Впрочем, споры насчет этого всё еще идут и микро-бенчмарки могут показывать очень разные результаты, зависящие от многих факторов. В clippy есть линт https://rust-lang.github.io/rust-clippy/master/#trivially_copy_pass_by_ref , но на данный момент он отключен из-за некоторых проблем с ложно-положительными срабатываниями.

Reply:
Ага, про Copy забыл, спасибо